web/src/stream_settings_ui.js

```diff
@@ -123,6 +123,9 @@ export function update_stream_name(sub, new_name) {
 
     // Update navbar if needed
     message_view_header.maybe_rerender_title_area_for_stream(sub);
+
+    // Update the create stream error if needed
+    stream_create.maybe_update_error_message();
```

Comment:
I think this will help with the bug Alya was facing. 

> After I renamed the archived stream, the error message ("An archived stream with this name already exists.") and renaming link did not disappear like they should.


Although I was never able to reproduce the bug, I think the bug could be fixed by updating the error message in server events.